pkg/teleporter/teleporter.go

```diff
@@ -0,0 +1,230 @@
+				return err
+			}
+		}
+		t.teleporterMessengerContractAddress = string(teleporterMessengerContractAddressBytes)
+	}
+	if t.teleporterMessengerDeployerAddress == "" {
+		var teleporterMessengerDeployerAddressBytes []byte
+		if utils.FileExists(teleporterMessengerDeployerAddressPath) {
+			teleporterMessengerDeployerAddressBytes, err = os.ReadFile(teleporterMessengerDeployerAddressPath)
+			if err != nil {
+				return err
+			}
+		} else {
+			// get teleporter deployer address
+			teleporterMessengerDeployerAddressBytes, err = utils.Download(teleporterMessengerDeployerAddressURL)
+			if err != nil {
+				return err
+			}
+			if err := os.MkdirAll(binDir, constants.DefaultPerms755); err != nil {
+				return err
+			}
+			if err := os.WriteFile(teleporterMessengerDeployerAddressPath, teleporterMessengerDeployerAddressBytes, constants.WriteReadReadPerms); err != nil {
+				return err
+			}
+		}
+		t.teleporterMessengerDeployerAddress = string(teleporterMessengerDeployerAddressBytes)
+	}
+	if t.teleporterMessengerDeployerTx == "" {
+		var teleporterMessengerDeployerTxBytes []byte
+		if utils.FileExists(teleporterMessengerDeployerTxPath) {
+			teleporterMessengerDeployerTxBytes, err = os.ReadFile(teleporterMessengerDeployerTxPath)
+			if err != nil {
+				return err
+			}
+		} else {
+			teleporterMessengerDeployerTxBytes, err = utils.Download(teleporterMessengerDeployerTxURL)
+			if err != nil {
+				return err
+			}
+			if err := os.MkdirAll(binDir, constants.DefaultPerms755); err != nil {
+				return err
+			}
+			if err := os.WriteFile(teleporterMessengerDeployerTxPath, teleporterMessengerDeployerTxBytes, constants.WriteReadReadPerms); err != nil {
+				return err
+			}
+		}
+		t.teleporterMessengerDeployerTx = string(teleporterMessengerDeployerTxBytes)
+	}
+	return nil
+}
+
+func (t *Deployer) Deploy(
+	teleporterInstallDir string,
+	version string,
+	subnetName string,
+	rpcURL string,
+	prefundedPrivateKey string,
+) (bool, string, string, error) {
+	alreadyDeployed, messengerAddress, err := t.DeployMessenger(teleporterInstallDir, version, subnetName, rpcURL, prefundedPrivateKey)
+	if err != nil {
+		return false, "", "", err
+	}
+	if alreadyDeployed {
+		return true, messengerAddress, "", nil
+	}
+	registryAddress, err := t.DeployRegistry(teleporterInstallDir, version, subnetName, rpcURL, prefundedPrivateKey)
+	if err != nil {
+		return false, "", "", err
+	}
+	return false, messengerAddress, registryAddress, nil
+}
+
+func (t *Deployer) DeployMessenger(
+	teleporterInstallDir string,
+	version string,
+	subnetName string,
+	rpcURL string,
+	prefundedPrivateKey string,
+) (bool, string, error) {
+	if err := t.DownloadAssets(teleporterInstallDir, version); err != nil {
+		return false, "", err
+	}
+	// check if contract is already deployed
+	client, err := evm.GetClient(rpcURL)
+	if err != nil {
+		return false, "", err
+	}
+	teleporterMessengerAlreadyDeployed, err := evm.ContractAlreadyDeployed(client, t.teleporterMessengerContractAddress)
```

Comment:
as `teleporterMessengerAlreadyDeployed` is not used anywhere else if can be reduced to 
```
if teleporterMessengerAlreadyDeployed, err :=; err!=nil
bla, return
else if teleporterMessengerAlreadyDeployed bla, return bla
```